### PR TITLE
#152: Persisting workspace before discovery

### PR DESCRIPTION
**Pull request by saurabh-prakash**
# Issue Link

#152 

# Description

Persisting workspace before discovery in order to avoid race condition as mentioned in attached issue

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Will be testing on stage TAS after triggering multiple jobs

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by palashmarantas — APPROVED** [review]

File: pkg/core/lifecycle.go [hunk 1]
```diff
@@ -236,6 +236,14 @@ func (pl *Pipeline) Start(ctx context.Context) (err error) {
 			return err
 		}
 
+		pl.Logger.Debugf("Caching workspace")
```
Comment (palashmarantas) [inline pkg/core/lifecycle.go:239]: why this change require ? persisting before discovery how does it help us ?

**Review by rishabh-arya95 — APPROVED** [review]

File: pkg/core/lifecycle.go (hunk 1 above)
Comment (saurabh-prakash) [reply pkg/core/lifecycle.go:239]: Please read the attached issue to get more context.
